pkg/lang/golang/plugin_expose.go

```diff
@@ -368,3 +398,158 @@ func (h *restAPIHandler) FindImports(f *core.SourceFile) (*sitter.Node, error) {
 	}
 	return imports, err
 }
+
+func (h *restAPIHandler) findChiRouterMounts(f *core.SourceFile, routerName string) []routerMount {
+	source := f.Program()
+	nextMatch := doQuery(f.Tree().RootNode(), findRouterMounts)
+	var mounts = make([]routerMount, 0)
+
+	for {
+		match, found := nextMatch()
+		if !found {
+			break
+		}
+
+		router_name, mount, path, package_name, package_func := match["router_name"], match["mount"], match["path"], match["package_name"], match["package_func"]
+
+		if !query.NodeContentEquals(router_name, f.Program(), routerName) ||
+			!query.NodeContentEquals(mount, f.Program(), "Mount") {
+			continue
+		}
+
+		mounts = append(mounts, routerMount{
+			Path:     stringLiteralContent(path, source),
+			PkgAlias: package_name.Content(source),
+			FuncName: package_func.Content(source),
+		})
+	}
+
+	return mounts
+}
+
+func (h *restAPIHandler) findChiRouterMountPackage(f *core.SourceFile, mount *routerMount) error {
+	source := f.Program()
+	nextMatch := doQuery(f.Tree().RootNode(), findImports)
+	for {
+		match, found := nextMatch()
+		if !found {
+			break
+		}
+
+		package_id, package_path := match["package_id"], match["package_path"]
+
+		if package_path == nil {
+			continue
+		}
+
+		p := strings.Split(stringLiteralContent(package_path, source), "/")
+		package_name := p[len(p)-1]
+		if package_id != nil {
+			if !query.NodeContentEquals(package_id, source, mount.PkgAlias) {
+				continue
+			}
+			mount.PkgName = package_name
+			return nil
+		}
+
+		if package_name == mount.PkgAlias {
+			mount.PkgName = package_name
+			return nil
+		}
+
+	}
+
+	return errors.Errorf("No import package found with name or alias [%s]", mount.PkgAlias)
```

Comment:
do we want to be returning errors from this plugin? Like at this point are we certain they meant to match the go chi syntax for us? If we ever supported another expose library for go would this get confusing?

Reply:
yeh i was kind of on the fence. I didn't put as many errors in the first time around since we could technically just silently continue if we don't find what we expect. At this point though, yeh we have determined that there is a mount call to an external package, but we cannot find the package in the imports. Though if this were to happen, the go build for the project just wouldn't work either.

Reply:
alright yeah thats fine then i guess, can always remove it